### PR TITLE
[SDP-1170] Add `PATCH /organization/circle-config` 

### PR DESCRIPTION
**Pull request by marwen-abid**
### What
* Add `PATCH /organization/circle-config` 

### Why
* Add the ability to configure Circle API Key and Wallet ID to be able to use the CircleSDK. 

#### PR Structure

* [x] This PR has a reasonably narrow scope (if not, break it down into smaller PRs).
* [x] This PR title and description are clear enough for anyone to review it.
* [x] This PR does not mix refactoring changes with feature changes (split into two PRs otherwise).

#### Thoroughness

* [x] This PR adds tests for the new functionality or fixes.
* [x] This PR contains the link to the Jira ticket it addresses.

#### Configs and Secrets

* [x] No new CONFIG variables are required -OR- the new required ones were added to the helmchart's [`values.yaml`] file.
* [x] No new CONFIG variables are required -OR- the new required ones were added to the deployments ([`pr-preview`], [`dev`], [`demo`], `prd`).
* [x] No new SECRETS variables are required -OR- the new required ones were mentioned in the helmchart's [`values.yaml`] file.
* [x] No new SECRETS variables are required -OR- the new required ones were added to the deployments ([`pr-preview secrets`], [`dev secrets`], [`demo secrets`], `prd secrets`).

#### Release

* [x] This is not a breaking change.
* [x] **This is ready for production.**. If your PR is not ready for production, please consider opening additional complementary PRs using this one as the base. Only merge this into `develop` or `main` after it's ready for production!

#### Deployment

* [ ] Does the deployment work after merging?

[`values.yaml`]: ../helmchart/sdp/values.yaml
[`pr-preview`]: https://github.com/stellar/kube/blob/d3e4f5dd8aa4c13b45a31a5a937f3e98841171a7/kube001-dev/namespaces/common-previews/stellar-disbursement-platform/backend-helm-values
[`dev`]: https://github.com/stellar/kube/blob/d3e4f5dd8aa4c13b45a31a5a937f3e98841171a7/kube001-dev/namespaces/stellar-disbursement-platform/backend-helm-values
[`demo`]: https://github.com/stellar/kube/blob/d3e4f5dd8aa4c13b45a31a5a937f3e98841171a7/kube001-dev/namespaces/stellar-disbursement-platform/demo/demo-backend-helm-values
[`pr-preview secrets`]: https://github.com/stellar/kube/blob/d3e4f5dd8aa4c13b45a31a5a937f3e98841171a7/kube001-dev/namespaces/common-previews/externalsecrets-common-previews.yaml#L241-L346
[`dev secrets`]: https://github.com/stellar/kube/blob/d3e4f5dd8aa4c13b45a31a5a937f3e98841171a7/kube001-dev/namespaces/stellar-disbursement-platform/stellar-disbursement-platform-externalsecrets.yaml
[`demo secrets`]: https://github.com/stellar/kube/blob/d3e4f5dd8aa4c13b45a31a5a937f3e98841171a7/kube001-dev/namespaces/stellar-disbursement-platform/demo/demo-sdp-externalsecrets.yaml

### CODE REVIEW

File: internal/serve/httphandler/circle_config_handler_test.go [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+func request(t *testing.T, r *chi.Mux, httpMethod string, body io.Reader) *httptest.ResponseRecorder {
+	t.Helper()
+
+	req, err := http.NewRequest(httpMethod, "/organization/circle-config", body)
+	require.NoError(t, err)
+
+	rr := httptest.NewRecorder()
+	r.ServeHTTP(rr, req)
+	return rr
+}
```
Comment (marcelosalloum) [inline internal/serve/httphandler/circle_config_handler_test.go:128]: It'd be nice to eventually move this to a package where other tests can import as well. Not something you need to do on thie PR though.

File: internal/circle/client_config_test.go [hunk 2]
```diff
@@ -0,0 +1,424 @@
...
+
+	return tx
+}
+
+func rollback(t *testing.T, dbTx db.DBTransaction) {
+	t.Helper()
+
+	err := dbTx.Rollback()
+	require.NoError(t, err)
+}
```
Comment (marcelosalloum) [inline internal/circle/client_config_test.go:424]: 💡  These are methods we could put in the dbtest.go file.

Comment (marcelosalloum) [reply internal/circle/client_config_test.go:424]: Maybe even a beginTx that rolls back automatically:

```suggestion
func beginTx(t *testing.T, ctx context.Context, dbConnectionPool db.DBConnectionPool, autoRollback bool) db.DBTransaction {
	t.Helper()

	tx, err := dbConnectionPool.BeginTxx(ctx, nil)
	require.NoError(t, err)
	
	if autoRollback {
		t.Cleanup(func() {
			rollback(t, dbTx)
		})
	}

	return tx
}

func rollback(t *testing.T, dbTx db.DBTransaction) {
	t.Helper()

	err := dbTx.Rollback()
	require.NoError(t, err)
}
```

**Review by marcelosalloum — APPROVED** [review]
LGTM! I've suggested a few polishings below, but nothing that would change the logic itself.

Plz, feel free to merge it after addressing the comments below.

File: internal/circle/client.go [hunk 3]
```diff
@@ -33,7 +33,7 @@ type ClientInterface interface {
 // Client provides methods to interact with the Circle API.
 type Client struct {
 	BasePath   string
-	APIKey     string
+	ApiKey     string
```
[the remark this answers is not in the document]
Comment (marwen-abid) [reply internal/circle/client.go]: Removed the commit with this change

File: internal/circle/client_config.go [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+	"errors"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/stellar/stellar-disbursement-platform-backend/db"
+)
+
+type ClientConfig struct {
+	EncryptedApiKey    *string   `db:"encrypted_api_key"`
```
[the remark this answers is not in the document]
Comment (marwen-abid) [reply internal/circle/client_config.go]: Addressed in 6688ab0aff7a64312bbbdf04f21fd52d4a4320fe

File: internal/circle/client_config_test.go (hunk 2 above)
Comment (marwen-abid) [reply internal/circle/client_config_test.go]: Nice, I like this suggestion 5a2b3119a9ccd6b55f17776d63e56cac176d898d

File: internal/serve/httphandler/circle_config_handler_test.go [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+	require.NoError(t, outerErr)
+	defer dbConnectionPool.Close()
+
+	kp := keypair.MustRandom()
+	encryptionPassphrase := kp.Seed()
+	encryptionPublicKey := kp.Address()
+
+	ccm := circle.ClientConfigModel{DBConnectionPool: dbConnectionPool}
+	encrypter := utils.DefaultPrivateKeyEncrypter{}
+	mDistributionAccountResolver := sigMocks.NewMockDistributionAccountResolver(t)
```
[the remark this answers is not in the document]
Comment (marwen-abid) [reply internal/serve/httphandler/circle_config_handler_test.go]: 8e71fead57351af8c5d5c6990dbf652334216625

File: internal/serve/httphandler/circle_config_handler_test.go (hunk 1 above)
Comment (marwen-abid) [reply internal/serve/httphandler/circle_config_handler_test.go]: addressed in 7a1753216fcf05918848edf80df7275c7527ddad

**Review by marcelosalloum — APPROVED** [review]
LGTM!

**Pull request merged**